app/jobs/timeout_stuck_builds_job.rb

```diff
@@ -0,0 +1,25 @@
+class TimeoutStuckBuildsJob < JobBase
+  @queue = :high
+
+  def self.perform
+    # check for builds in runnable that are no longer in the queue
+    missing = []
+    BuildAttempt.select("build_attempts.id", " build_parts.queue as queue").joins(:build_part)
+                .where("build_attempts.state = 'runnable' AND build_attempts.created_at < ? AND build_attempts.created_at > ?", 5.minutes.ago, 1.day.ago)
+                .group_by(&:queue)
+                .each do |queue, attempts|
+                  current_queue = Resque.redis.lrange("queue:#{queue}", 0, -1).to_s
+                  missing += attempts.reject { |attempt| current_queue.match(/build_attempt_id\\*\"\:#{attempt.id}[^0-9]/) }
+                end
+
+    missing.select { |build_attempt_partial| BuildAttempt.find(build_attempt_partial.id).state == 'runnable' }
```

Comment:
you need to use `select!` here. Or chain the following `.each` off of this select call.